qdrant-landing/content/articles/vector-search-production.md

```diff
@@ -0,0 +1,517 @@
+
+### How to Predict and Test Cluster Performance
+
+||
+|:-:|
+|**"I had no idea something was wrong until it was too late."**|
+
+Such issues tend to occur when users don't monitor resource usage, search performance, or security. Critical failures went undetected until users complained.
+
+✅ **Run load tests** under expected traffic conditions to identify bottlenecks before go-live.
+
+You need to set a plan to use realistic data for your load tests. Ideally, you should test with production traffic or historical data that closely resembles your actual workload. This provides more accurate results than randomly generated test data, as it will better represent real-world usage patterns and data distributions.
+
+> Design your load test to gradually increase traffic until you reach or exceed your expected production load. For example, if you expect 1000 requests per second (RPS) in production, incrementally scale up your test to reach this threshold while monitoring latency. Responses will separately show server timing for granular monitoring.
+
+You should test system performance after restarts to understand cold-start behavior. Initial queries after a restart may be significantly slower as caches need to be rebuilt. For example, a query might take 50-60 seconds initially but only 0.5 seconds on subsequent runs.
+
+Remember, cold-starts and query behaviour are dataset dependent, which is why you should establish your own baselines and what to expect.
+
+||
+|-|
+|**Read More:** [Distributed Deployment Documentation](https://qdrant.tech/documentation/guides/distributed_deployment/)
+
+### How to Design Your Systems to Protect Against Failure
+
+||
+|:-:|
+|**"When one node crashed, our entire cluster went down."**|
+
+Unfortunately, you didn't plan for failover or chaos testing, so a single point of failure took out production. You should plan for hardware or node crashes by storing data redundantly, testing failovers, and running chaos experiments.
+
+✅ **Regularly test failures** to reveal how your system recovers.
+
+High concurrency and large memory footprints can expose misconfigurations more quickly, so regularly simulating failures reveals how your system recovers.
+
+- **Graceful Node Shutdowns**: Drain queries, reassign shard ownership via load balancer.
+- **Redundant Data Paths**: Store data on multiple volumes or in multiple locations.
+- **Load Tests**: Generate high concurrency or large queries to mimic real surge patterns.
+
+### Set up Telemetry for Early Detection
+
+❓ **Use Case:** A customer in health care uses telemetry data from their Qdrant deployment to identify performance and scaling issues with their open-source implementation. The telemetry helps them monitor metrics such as search performance, RAM utilization efficiency, and indexing speed. By analyzing this data, they can work toward reducing query response times from 50-60 seconds to 0.5 seconds and optimize their system configuration.
```

Comment:
It sounds awful that someone can initially have 50-60 seconds of latency. Is that a real case? Also, 0.5 seconds is sometimes too much. I remember the last call with one of the users, and they reported less than 80ms.